### PR TITLE
Remove permission config.

### LINKED ISSUE #2199: Remove permissions config
Follow-up to #2089 (lock down kv endpoints) and needed for #2090 (make configs aware of structured data layout).

Permissions are no longer adjustable on the KV endpoint: system users only (root and node).

### PR DESCRIPTION
**Pull request by mberhault**
Fixes #2199

Mostly code removal. Some tests (eg: storage/replicate_test.go) used
the permission config to test gossip. Changed to use other configs.

I have kept the Read/Write flags on requests, they may still come in
handy.

For info: TestRangeNoGossipConfig and TestRangeNoGossipFromNonLeader from storage/replica_test.go still fail. I'm trying to figure out why that is exactly. No luck so far.

### CODE REVIEW

File: kv/dist_sender.go [hunk 1]
```diff
@@ -197,73 +196,16 @@ func NewDistSender(ctx *DistSenderContext, gossip *gossip.Gossip) *DistSender {
...
 // verifyPermissions verifies that the requesting user (header.User)
-// has permission to read/write (capabilities depend on method
-// name). In the event that multiple permission configs apply to the
-// key range implicated by the command, the lowest common denominator
-// for permission. For example, if a scan crosses two permission
-// configs, both configs must allow read permissions or the entire
-// scan will fail.
+// is allowed to perform the requested operations.
+// All KV endpoints are restricted to system users, with 'root'
+// being the value or arg.User.
```
Comment (tamird) [inline kv/dist_sender.go]: s/or/of/

File: storage/store.go [hunk 2]
```diff
@@ -894,15 +892,6 @@ func (s *Store) BootstrapRange() error {
 	if err := engine.MVCCPutProto(batch, ms, key, now, nil, acctConfig); err != nil {
 		return err
 	}
-	// Permission config.
-	permConfig := &config.PermConfig{
-		Read:  []string{security.RootUser}, // root user
```
Comment (tamird) [inline storage/store.go]: this is reason `TestRangeNoGossipConfig` aren't passing now - that they were passing before was a lie due to this default config matching exactly the config in the test. The problem is a race between the test writing the new config and this method running (which publishes the configs into gossip).

**Comment by MBerhault** [comment]
ok, so the tests are failing because the config we write at the beginning of the test (initConfigs) do not get gossiped. This can be verified by changing TestRangeGossipAllConfigs to:
```
  for _, test := range testData {
    info, err := tc.gossip.GetInfo(test.gossipKey)
    if err != nil {
      t.Fatal(err)
    }
    configMap := info.(config.PrefixConfigMap)
    expConfigs := test.configs
    if !reflect.DeepEqual([]config.PrefixConfig(configMap), expConfigs) {
      t.Errorf("expected gossiped configs to be equal %s vs %s", configMap, expConfigs)
    }
  }
```

Looks like we should force gossiping before writing any new values. Otherwise, we're expecting the configs from `initConfigs` rather than the defaults in `store.go`

File: kv/dist_sender_test.go [hunk 3]
```diff
@@ -711,42 +686,40 @@ func TestVerifyPermissions(t *testing.T) {
 		hasPermission    bool
 	}{
 		// Test permissions within a single range
-		{readOnlyRequests, "read1", proto.KeyMin, proto.KeyMin, true},
-		{readOnlyRequests, "rw1", proto.KeyMin, proto.KeyMin, true},
-		{readOnlyRequests, "write1", proto.KeyMin, proto.KeyMin, false},
+		{readOnlyRequests, "root", proto.KeyMin, proto.KeyMin, true},
+		{readWriteRequests, "root", proto.KeyMin, proto.KeyMin, true},
```
Comment (tamird) [inline kv/dist_sender_test.go]: should these instances of `"root"` be `security.RootUser`?

**Comment by tamird** [comment]
LGTM

**Comment by petermattis** [comment]
So much code deletion. LGTM.

**Comment by tamird** [comment]
@MBerhault pushed a commit that fixes those two tests.

File: kv/dist_sender.go (hunk 1 above)
Comment (MBerhault) [inline kv/dist_sender.go]: Done.

File: kv/dist_sender_test.go (hunk 3 above)
Comment (MBerhault) [inline kv/dist_sender_test.go]: oops. Done.

**Pull request merged**